AutoGear.lua

```diff
@@ -4247,10 +4321,17 @@ function AutoGearTooltipHook(tooltip)
 		)
 	end
 end
-GameTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
-ShoppingTooltip1:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
-ShoppingTooltip2:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
-ItemRefTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
+
+if TOC_VERSION_CURRENT < 100002 then
+	GameTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
+	ShoppingTooltip1:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
+	ShoppingTooltip2:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
+	ItemRefTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
+else
+	-- For Dragonflight and beyond, registering callbacks with the new TooltipDataProcessor.AddTooltipPostCall function is the way to hook tooltips.
+	-- Source: https://warcraft.wiki.gg/wiki/Patch_10.0.2/API_changes#Tooltip_Changes
+	TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Item, AutoGearTooltipHook)
+end
```

Comment:
I think the Dragonflight-specific fixes should be removed from this pull request and separated into a different pull request.  That would require having the changes in a different branch in your fork.

Reply:
drop the tostring